src/test/run-make/tools.mk

```diff
@@ -80,12 +99,20 @@ REMOVE_RLIBS      = rm $(TMPDIR)/$(call RLIB_GLOB,$(1))
 
 %.a: %.o
 	ar crus $@ $<
+%.lib: lib%.o
+	ar crus $@ $<
 %.dylib: %.o
 	$(CC) -dynamiclib -Wl,-dylib -o $@ $<
 %.so: %.o
 	$(CC) -o $@ $< -shared
+
+ifdef IS_MSVC
+%.dll: lib%.o
+	$(CC) $< -link -dll -out:$@
```

Comment:
Ah it looks like this needs the same `cygpath -w` treatment as some above flags, wheeeee